src/crypto/tests/TestPSAOpKeyStore.cpp

```diff
@@ -0,0 +1,219 @@
+    NL_TEST_ASSERT(inSuite, err == CHIP_ERROR_INVALID_SIGNATURE);
+
+    // Committing with mismatching fabric fails, leaves pending
+    err = opKeystore.CommitOpKeypairForFabric(kBadFabricIndex);
+    NL_TEST_ASSERT(inSuite, err == CHIP_ERROR_INVALID_FABRIC_INDEX);
+    NL_TEST_ASSERT(inSuite, opKeystore.HasPendingOpKeypair() == true);
+    NL_TEST_ASSERT(inSuite, opKeystore.HasOpKeypairForFabric(kFabricIndex) == true);
+
+    // Committing key resets pending state
+    err = opKeystore.CommitOpKeypairForFabric(kFabricIndex);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, opKeystore.HasPendingOpKeypair() == false);
+    NL_TEST_ASSERT(inSuite, opKeystore.HasOpKeypairForFabric(kFabricIndex) == true);
+
+    // After committing, signing works with the key that was pending
+    P256ECDSASignature sig3;
+    uint8_t message2[] = { 10, 11, 12, 13 };
+    err                = opKeystore.SignWithOpKeypair(kFabricIndex, ByteSpan{ message2 }, sig3);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    err = csrPublicKey2.ECDSA_validate_msg_signature(message2, sizeof(message2), sig3);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    // Let's remove the opkey for a fabric, it disappears
+    err = opKeystore.RemoveOpKeypairForFabric(kFabricIndex);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, opKeystore.HasPendingOpKeypair() == false);
+    NL_TEST_ASSERT(inSuite, opKeystore.HasOpKeypairForFabric(kFabricIndex) == false);
+}
+
+void TestEphemeralKeys(nlTestSuite * inSuite, void * inContext)
+{
+    PSAOperationalKeystore opKeyStore;
+
+    Crypto::P256ECDSASignature sig;
+    uint8_t message[] = { 'm', 's', 'g' };
+
+    Crypto::P256Keypair * ephemeralKeypair = opKeyStore.AllocateEphemeralKeypairForCASE();
+    NL_TEST_ASSERT(inSuite, ephemeralKeypair != nullptr);
+    NL_TEST_ASSERT_SUCCESS(inSuite, ephemeralKeypair->Initialize(Crypto::ECPKeyTarget::ECDSA));
+
+    NL_TEST_ASSERT_SUCCESS(inSuite, ephemeralKeypair->ECDSA_sign_msg(message, sizeof(message), sig));
+    NL_TEST_ASSERT_SUCCESS(inSuite, ephemeralKeypair->Pubkey().ECDSA_validate_msg_signature(message, sizeof(message), sig));
+
+    opKeyStore.ReleaseEphemeralKeypair(ephemeralKeypair);
+}
+
+/**
+ *   Test Suite. It lists all the test functions.
+ */
+static const nlTest sTests[] = { NL_TEST_DEF("Test Basic Lifecycle of PersistentStorageOperationalKeystore", TestBasicLifeCycle),
+                                 NL_TEST_DEF("Test ephemeral key management", TestEphemeralKeys), NL_TEST_SENTINEL() };
+
+/**
+ *  Set up the test suite.
+ */
+int Test_Setup(void * inContext)
+{
+    CHIP_ERROR error = chip::Platform::MemoryInit();
+    VerifyOrReturnError(error == CHIP_NO_ERROR, FAILURE);
+
+#if CHIP_CRYPTO_PSA
+    psa_crypto_init();
+#endif
+
+    return SUCCESS;
+}
+
+/**
+ *  Tear down the test suite.
+ */
+int Test_Teardown(void * inContext)
+{
```

Comment:
no crypto release here ? to undo psa_crypto_init

Reply:
There seems to be no API for that: https://armmbed.github.io/mbed-crypto/html/api/library/library.html.